### PR TITLE
Prometheus: adding instance and mtls

### PR DESCRIPTION
**Pull request by mcmikemn**
I'm working on adding instance and mtls to Prometheus.

In my local .env, Prometheus was working in my VPN. The only things I changed are commited here, but now I get "Bad Gateway". I'm testing with Prometheus in my VPN but I don't have any sentry authentication in front of it.

There don't appear to be any errors in Prometheus' logs, but this is in Traefik's log:
```
traefik-1      | time="2024-06-06T01:04:43Z" level=error msg="sourceRange is empty, IPAllowLister not created" entryPointName=websecure routerName=prometheus-default-prometheus@docker
```
But I have `PROMETHEUS_METRICS_IP_SOURCERANGE="0.0.0.0/0"` in my .env and Traefik seems to see the IPAllowList value:
image

### CODE REVIEW

**Comment by mcmikemn** [comment]
@EnigmaCurry, I think this works and is ready to merge.

File: prometheus/Makefile [hunk 1]
```diff
@@ -30,11 +45,11 @@ compose-profiles:
 shell:
 	@container=$$(eval "${BIN}/script-wizard choose 'docker exec -it into which container?' 'node-exporter' 'prometheus' 'grafana' 'cadvisor' 'alertmanager' 'alert-test' --default 'grafana'") && make --no-print-directory docker-compose-shell SERVICE=$${container}
 
-.PHONY: local-alertmanager
+.PHONY: local-alertmanager # need summary here
```
Comment (EnigmaCurry) [inline prometheus/Makefile:48]: can you add the summary here or remove the comment altogether

File: prometheus/Makefile [hunk 2]
```diff
@@ -30,11 +45,11 @@ compose-profiles:
...
 local-alertmanager:
 	@export ENV_FILE=${ENV_FILE} CONTEXT_INSTANCE=${CONTEXT_INSTANCE} PORT_FORWARD="$$(${BIN}/port_forward ${ENV_FILE} alertmanager 9093)"; if [[ ! "$${PORT_FORWARD}" =~ [0-9]+ ]]; then ${BIN}/fault "Could not get the SSH forwarded port"; else echo "Local port forward established on localhost:$${PORT_FORWARD}"; fi;
 
-.PHONY: test-alert
+.PHONY: test-alert # need summary here
```
Comment (EnigmaCurry) [inline prometheus/Makefile:52]: can you add the summary here or remove the comment altogether

File: prometheus/docker-compose.instance.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      - "traefik.http.middlewares.mtlsauth-(@= router @).plugin.certauthz.domains=(@= mtls_authorized_certs @)"
+        #@ enabled_middlewares.append("mtlsauth-{}".format(router))
+        #@ end
+        #@ enabled_middlewares.append("mtls-header@file")
+      #@ end
+      
+      #! Override the default port that the app binds to:
+      #! You don't normally need to do this, as long as your image has
+      #! an EXPOSE directive in it, Traefik will autodetect it, but this is how you can override it:
+      - "traefik.http.services.(@= router @).loadbalancer.server.port=8000"
```
Comment (EnigmaCurry) [inline prometheus/docker-compose.instance.yaml:62]: Wasn't this supposed to be removed or set to 3000?

**Review by EnigmaCurry — CHANGES_REQUESTED** [review]

**Comment by EnigmaCurry** [comment]
I think maybe you forgot to push.

File: prometheus/Makefile (hunk 2 above)
Comment (mcmikemn) [reply prometheus/Makefile]: fixed in [5c113f7](https://github.com/EnigmaCurry/d.rymcg.tech/pull/234/commits/5c113f7b949509d3e6f63cc83f7fbcdf9a581a14)

File: prometheus/docker-compose.instance.yaml (hunk 3 above)
Comment (mcmikemn) [reply prometheus/docker-compose.instance.yaml]: It was removed in [ea9b97b](https://github.com/EnigmaCurry/d.rymcg.tech/pull/234/commits/ea9b97b21c1d51310e3a36697dc185fbb57e2258)

**Comment by mcmikemn** [comment]
> I think maybe you forgot to push.

Hmmm, I see the changes I pushed a few dayes ago here in the PR.

**Review by EnigmaCurry — APPROVED** [review]

**Pull request merged**